pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -224,7 +224,7 @@ type PipelineTask struct {
 func (et *EmbeddedTask) IsCustomTask() bool {
 	// Note that if `apiVersion` is set to `"tekton.dev/v1beta1"` and `kind` is set to `"Task"`,
 	// the reference will be considered a Custom Task - https://github.com/tektoncd/pipeline/issues/6457
-	return et != nil && et.APIVersion != "" && et.Kind != ""
+	return et != nil && et.APIVersion != "" && et.Kind != "" && et.Kind != string(ClusterTaskKind)
```

Comment:
I wonder if we check (et.APIVersion!="v1beta1"&&et.Kind != string(ClusterTaskKind)) would be better?

Reply:
ah don't think we should handle `ClusterTask` differently from `Task` -- maybe what we need is better documentation?